#!/usr/bin/env python
PACKAGE = "motor_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("inverted",         bool_t,     0,  "Inverted",         False)
gen.add("peak_voltage",     double_t,   0,  "Peak Voltage",     12.0,   0.0,    28)
gen.add("pot",              bool_t,     0,  "Potentiometer",    False)
gen.add("invert_sensor",    bool_t,     0,  "Invert Sensor",    False)
gen.add("cont_current",     int_t,      0,  "Continuous Current Limit (A)", 30,     0,  60)
#gen.add("peak_current",     int_t,      0,  "Peak Current Limit (A)",       30,    0,  100)
gen.add("peak_current_dur", int_t,      0,  "Peak Current Duration (ms)",   100,    0,  2000)
gen.add("brake_mode",       bool_t,     0,  "Brake Mode",       True)

gen.add("P", double_t, 0, "P", 0, 0, 100000)
gen.add("I", double_t, 0, "I", 0, 0, 100000)
gen.add("D", double_t, 0, "D", 0, 0, 100000)
gen.add("F", double_t, 0, "F", 0, 0, 100000)

exit(gen.generate(PACKAGE, "motor_control", "Talon"))